asv_benchmarks/benchmarks/cluster.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+
+from sklearn.cluster import KMeans
+from sklearn.cluster._kmeans import _k_init
+from sklearn.utils.extmath import row_norms
+
+from .common import Benchmark, Estimator, Predictor, Transformer
+from .datasets import _blobs_dataset, _20newsgroups_highdim_dataset
+from .utils import neg_mean_inertia
+
+
+class KMeans_bench(Predictor, Transformer, Estimator, Benchmark):
+    """
+    Benchmarks for KMeans.
+    """
+
+    param_names = ['representation', 'algorithm', 'n_jobs']
```

Comment:
n_jobs should be removed. It's a bit worrying that the inconsistency between this and `params` is not picked up anywhere

Reply:
Yes, it's weird that asv does not check length consistency of params and param_names.

Reply:
good catch  btw :)